src/endpoints/onboarding_endpoints.py

```diff
@@ -23,11 +26,83 @@ def __init__(self, client: Steamship, agent_service: AgentService):
         self.client = client
         self.agent_service = agent_service
 
+    @post("/onboarding/character/name")
+    def set_character_name(self, name: str, context_id: Optional[str] = None, **kwargs):
+        context = self.agent_service.build_default_context(
+            context_id=context_id, **kwargs
+        )
+
+        if not _is_allowed_by_moderation(name, context):
+            raise SteamshipError(
+                "Supplied 'name' was rejected by game's moderation filter. Please try again."
+            )
+
+        game_state = get_game_state(context)
+        game_state.player.name = name
+        save_game_state(game_state, context)
+
+    @post("/onboarding/character/background")
+    def set_character_background(
+        self, background: str, context_id: Optional[str] = None, **kwargs
+    ):
+        context = self.agent_service.build_default_context(
+            context_id=context_id, **kwargs
+        )
+
+        if not _is_allowed_by_moderation(background, context):
+            raise SteamshipError(
+                "Supplied 'background' was rejected by game's moderation filter. Please try again."
+            )
+
+        game_state = get_game_state(context)
+        game_state.player.background = background
+        save_game_state(game_state, context)
+
+    @post("/onboarding/character/description")
+    def set_character_description(
+        self,
+        description: str,
+        update: Optional[bool] = True,
+        context_id: Optional[str] = None,
+        **kwargs,
+    ):
+        context = self.agent_service.build_default_context(
+            context_id=context_id, **kwargs
+        )
+
+        if not _is_allowed_by_moderation(description, context):
+            raise SteamshipError(
+                "Supplied 'description' was rejected by game's moderation filter. Please try again."
+            )
+
+        game_state = get_game_state(context)
+        game_state.player.description = description
+
+        if game_state.player.description and game_state.player.name:
+            if (not game_state.image_generation_requested()) or update:
+                if image_gen := get_image_generator(context):
+                    task = image_gen.request_profile_image_generation(context=context)
+                    character_image_block = task.wait().blocks[0]
+                    game_state.player.profile_image_url = (
+                        character_image_block.raw_data_url
+                    )
+                    game_state.profile_image_url = character_image_block.raw_data_url
+                    save_game_state(game_state, context)
+
+        save_game_state(game_state, context)
+
     @post("/complete_onboarding")
-    def complete_onboarding(self, **kwargs) -> bool:
+    def complete_onboarding(self, context_id: Optional[str] = None, **kwargs) -> bool:
```

Comment:
suggestion[non-blocking]: If you peek into `build_default_context`, the context id is always ignored, all it does is log a warning if you pass it.  I would vote to stop accepting it as a parameter altogether if we're not actually going to do anything with it. 

This applies to all of the endpoints in this class.

Reply:
curious: I'm not sure I fully understand the decision to hardcode the context ID. Is that simply because we don't want multiple games in the same workspace? That feels like a binding that is not strictly necessary and would function the same with default values when not provided as it does with hardcoding.

Ignoring context_id departs from the SDK and that makes me nervous only in the way we are training ourselves around certain assumptions.